controllers/storagecluster/consumer.go

```diff
@@ -0,0 +1,168 @@
+package storagecluster
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"google.golang.org/grpc"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/types"
+
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v1"
+	providerClient "github.com/red-hat-storage/ocs-operator/services/provider/client"
+	"github.com/red-hat-storage/ocs-operator/services/provider/common"
+)
+
+var (
+	// externalOCSResources will hold the ExternalResources for storageclusters
+	// ExternalResources can be accessible using the UID of an storagecluster
+	// getExternalConfigFromProvider func sets value for this once and use same again
+	externalOCSResources = map[types.UID][]ExternalResource{}
+)
+
+type ocsConsumer struct{}
+
+// ensureCreated performs onboarding and update capacity
+func (o *ocsConsumer) ensureCreated(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
+
+	if !isExternalOCSProvider(instance) {
+		return nil
+	}
+
+	if instance.Status.ExternalStorage.ConsumerID == "" {
+		return onBoardConsumer(r, instance)
+	}
+
+	if instance.Spec.ExternalStorage.RequestedCapacity != nil &&
```

Comment:
How can a consumer be onboarded and not have RequestedCapacity?

Reply:
As per the design we are taking 1TB as a default value if a consumer is not specifying any value.